climbing_stairs/climbing_stairs.py

```diff
@@ -2,8 +2,52 @@
 
 import sys
 
+#Built dict determining the base number of whole actions possible for each option of 3,2,1 respectively 
+#Next step is to iterate over n - base number of whole actions and the subsequent permutations to get n to 0 
+#if options[3] > 0, then n - (3 * permutation(options[3])) and then recurse this until n = 0 for each option[x] reduction 
+#keep dict of permutations and check to see if it's been called already via memoizer
+
 def climbing_stairs(n, cache=None):
-  pass 
+    if n < 0:
+      return 0
+    elif n == 0 or n == 1:
+      return 1
+    elif n == 2:
+      return 3
```

Comment:
This needs to be a 2.  if you have 2 steps you only have 2 options for going up those steps.`1 + 1` or `2`